### DISCUSSION TITLE
Rename Programs IN PLACE. It currently creates duplicates.

### DISCUSSION THREAD

**Discussion by troach242** [christofmuc/KnobKraft-orm#339] [Ideas]
I own an Oberheim Matrix 1000 that I want to use your librarian with. The Matrix 1000 ignores program names (silly oversight on Oberheim's part) so in dumps, your librarian correctly names them crypticly. I created lists of the REAL names of the programs and tried to bulk rename. This seems to work, but I end up with duplicates that I am unable to delete (other than 1 by 1)... Can we have RENAME in place implemented.

Also - bulk deletes would be fantastic... I sure wish I could select multiple programs to delete in one pass. :)

Let me know how I can help. I would be happy to test things for you where I can.
Thanks.
Thom

**Comment by christofmuc** [comment]
Bulk deletes are possible in that you can always delete everything that matches the current filter. E..g you can delete a whole import, or all patches with a specific tag that you  have just filtered. If you select Edit... Delete Patches... via the menu it will offer to delete everything in the current grid, even if it is the whole database (it'll ask for confirmation).

The Matrix 1000 is really special, and at some point I had it working perfectly (I have one as well), but things might break with rising complexity.

So... First question is which version you are using, and if you are using the Matrix adapation or the old built-in system. (It'll have a tab adaptation when you have an adaptation synth selected, else there is a settings tab).

The idea is that we tell the software which names are "default", and therefore are inferior to given names. The Matrix 1000 default names are BNKn: nnn, and should never overwrite "better" names. If you reimport a patch with a name that is not considered a default name, it will rename the patch with the newly imported name.

Patch duplication only happens when the fingerprinting of the patches incorrectly not ignores the stored name. I actually have to look into the implementation, as this is one of the older parts of the whole codebase.

Can you describe the workflow you do and when the problem starts occuring in a bit more detail?

**Reply by troach242** [reply]
Ha.. The deletes work very well. I was confused by the drag and drop can and did not explore the [edit][delete patches] option. that is exactly what I was looking for.
As the the Matrix-1000, I was using the python adaptation until I saw your reply and tried the native adaptation. Rename is working as it should now.
I see about the odd matrix-1000 name thing. That was silly on Oberheim's part considering the sysex compatibility with the Matrix-6. Why couldnt they just store it. On the bright side, I generally load directly from my computer when composing, and dont need to reload 0-199 into the librarian. I just have so many programs for the matrix-1000 that I really needed your storage, searching, tagging and sorting the most. 

Thank you again for the super quick support. This is an amazing product. I am starting to ramp up use for seven synthesizers. It is a dream to open one librarian and find what you need fast.

**Reply by christofmuc** [reply]
Great, glad you like it! I'll log this as a bug in the Matrix 1000 adaptation then, because effectively at some point I want to get rid of the special code (that predates the whole adaptation mechanism). On the other hand, those were the synths that I had first and implemented first, and they are probably the most complete implementations...
